onnxruntime/core/providers/cuda/math/softmax.cc

```diff
@@ -26,7 +26,7 @@ Status SoftMaxComputeHelper(
 
   // cudnnSoftmaxForward/Backward is not optimal implementation.
   // TODO: remove cudnn path completely in the future.
-  if (D == input_shape[normalized_axis] && D <= 1024 && D * sizeof(T) <= 4096) {
+  if (D <= 1024 && D * sizeof(T) <= 4096) {
```

Comment:
I think we can relax the "D == input_shape[normalized_axis]" condition... 
But let's also add some test cases to cover this .

Reply:
The existing tests are invoking this case (i.e. D != input_shape[normalized_axis]) since shape is {8, 2, 128} and an unspecified axis defaults to one. But I can add a different shape/axis combination.

Reply:
good point... I though the default axis is -1 :)

Reply:
Add tests for all axes (considering last dimension as special case).

Reply:
Oh, I missed your comment. Ha :)